### PR TITLE
Bump to Stackage LTS 20.26

### PR DESCRIPTION
**Pull request by andreasabel**
- **Cabal files updated for 0.9.5.0**
- **Some missing golden values**
- **Bump stack.yaml to LTS 20.26 (GHC 9.2.8)**
   Removes some `extra-deps`
- **Fix CI caching**

CI with caching takes now ~5 min rather than ~20 min.  
- https://github.com/andreasabel/granule/actions/runs/8540861693/job/23398869273?pr=1

It still fails because of:
- #240 
- #241

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -40,8 +34,9 @@ extra-package-dbs: []
 #
 # Extra directories used by stack for building
 
-extra-include-dirs:
-- /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi
+## Superfluous?
+# extra-include-dirs:
+# - /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi
```
Comment (buggymcbugfix) [inline stack.yaml:39]: @dorchard git blame says you added this in 2022. Can we remove it?

File: .github/workflows/haskell.yml [hunk 2]
```diff
@@ -15,35 +15,55 @@ jobs:
...
+    #   run: stack build
+
+    - name: Build w/ tests
+      run: stack test --no-run-tests
 
     - name: Run tests
       run: stack test
+
+    # Always cache dependencies.
+    # This makes esp. sense now that some tests are broken.
```
Comment (buggymcbugfix) [inline .github/workflows/haskell.yml:61]: Can you clarify this comment? I am not up-to-date on best practices.

I am working on fixing the tests, so once they are fixed, should we not cache dependencies?

**Review by buggymcbugfix — COMMENTED** [review]
Hi @andreasabel, thank you for kicking off the granule spring clean!

File: .github/workflows/haskell.yml (hunk 2 above)
Comment (andreasabel) [reply .github/workflows/haskell.yml:61]: No, sure.  
It is just that the secret sauce `always()` will enable caching _even_ if a previous step failed (such as building or running).

Comment (buggymcbugfix) [reply .github/workflows/haskell.yml:61]: Ah, that makes a lot of sense, thanks.

File: stack.yaml (hunk 1 above)
Comment (dorchard) [reply stack.yaml:39]: I believe so. Is this redundant for building on recent mac OS X?

Comment (buggymcbugfix) [reply stack.yaml:39]: It builds fine on my 2020 M1 under macOS 14.4.1

Comment (dorchard) [reply stack.yaml:39]: Looks like we don't need it then.

**Pull request merged**